editor/plugins/org.fusesource.ide.project/src/org/fusesource/ide/project/camel/CamelFacetInstallationDelegate.java

```diff
@@ -0,0 +1,245 @@
+				createBlueprintDescriptor(osgiInf);
+			} else if (dsl.equalsIgnoreCase("Spring")) {
+				createSpringDescriptor(metaInf);
+			} else if (dsl.equalsIgnoreCase("Routes")) {
+				createRoutesDescriptor(metaInf);
+			}
+			createUtilityManifest(project, metaInf, new NullProgressMonitor());
+		}
+	}
+	protected IPath[] getAllSourceMappings(IProject p) {
+		ComponentResource[] res = findAllMappings(p);
+		IPath[] sourcePaths = new IPath[res.length];
+		for( int i = 0; i < res.length; i++ ) {
+			sourcePaths[i] = res[i].getSourcePath();
+		}
+		return sourcePaths;
+	}
+	
+	protected ComponentResource[] findAllMappings(IProject project) {
+		StructureEdit structureEdit = null;
+		try {
+			structureEdit = StructureEdit.getStructureEditForRead(project);
+			WorkbenchComponent component = structureEdit.getComponent();
+			Object[] arr = component.getResources().toArray();
+			ComponentResource[] result = new ComponentResource[arr.length];
+			for( int i = 0; i < arr.length; i++ )
+				result[i] = (ComponentResource)arr[i];
+			return result;
+		} catch (NullPointerException e) {
+			// TODO log
+		} finally {
+			if(structureEdit != null)
+				structureEdit.dispose();
+		}
+		return new ComponentResource[]{};
+	}
+	
+	private IFolder findFolder(IContainer[] p, final String name) {
+		final IFolder[] found = new IFolder[1]; 
+		
+		for( int i = 0; i < p.length && found[0] == null; i++ ) {
+			try {
+				p[i].accept(new IResourceVisitor() {
+					public boolean visit(IResource resource) throws CoreException {
+						if( resource instanceof IFolder && resource.getName().equals(name))
+							found[0] = (IFolder)resource;
+						return found[0] == null;
+					}
+				});
+			} catch(CoreException ce) {
+			}
+		}
+		return found[0];
+	}
+	
+	
+	private void createUtilityProjectReorganized(IProject project) throws CoreException{
+		String strContentFolder = model.getStringProperty(ICamelFacetDataModelProperties.CAMEL_CONTENT_FOLDER);
+		project.setPersistentProperty(ICamelFacetDataModelProperties.QNAME_CAMEL_CONTENT_FOLDER, strContentFolder);
+		
+		IFolder camelContent = project.getFolder(strContentFolder);
+		IProgressMonitor monitor = new NullProgressMonitor();
+		
+		IVirtualComponent newComponent = ComponentCore.createComponent(project);
+		final IVirtualFolder jbiRoot = newComponent.getRootFolder();
+
+		// Map the CAMELcontent to root for deploy
+		String resourcesFolder = model.getStringProperty(ICamelFacetDataModelProperties.CAMEL_CONTENT_FOLDER);
+		jbiRoot.createLink(new Path("/" + resourcesFolder), 0, null); //$NON-NLS-1$
+
+		IFolder metainf = project.getFolder("src").getFolder("META-INF"); 
+		if( metainf.exists())
+			metainf.move(camelContent.getFolder("META-INF").getFullPath(), true, new NullProgressMonitor());
+		createUtilityManifest(project, camelContent.getFolder("META-INF"), monitor);
+		
+		String dsl = model.getStringProperty(ICamelFacetDataModelProperties.CAMEL_DSL);
+		if( dsl.equalsIgnoreCase("Blueprint") ) {
```

Comment:
seems duplicated code with addElementsToUtilityProject
Just a parameter is different